### ISSUE TITLE
[css-logical-1] spec should introduce propdef table notation for logical properties

### ISSUE THREAD

**Issue by dbaron** [w3c/csswg-drafts#2822]
The [section on Box Model Properties](https://drafts.csswg.org/css-logical-1/#box) introduces a large number of new properties which are very different from all other CSS properties in that they don't have independent computed values, because their computed values mirror another CSS property.  While this is expressed in the prose, it seems like it should be expressed in the propdef table, since other specifications will (hopefully) be introducing similar properties, and it would be good to have a consistent notation for introducing them rather than doing so with prose that may come over time to have normative differences.

I got here from w3ctag/design-reviews#286.

**Comment by dbaron** [comment]
Also, I think there's a decent case to be made that the relevant propdef table entry should say both:
* which element's writing mode is used for the logical mapping, and
* which physical properties are being mapped to.

**Comment by Loirooriol** [comment]
> other specifications will (hopefully) be introducing similar properties

This is already happening, e.g. https://drafts.csswg.org/css-overflow-3/#logical

**Comment by fantasai** [comment]
For properties, the mapping is always using the writing mode of the element itself. So that shouldn't need to go into the propdef table.

**Comment by fantasai** [comment]
If we do this, rather than adding a statement like @Loirooriol proposes in https://github.com/w3c/csswg-drafts/issues/3033 then I think it would be something like:

```
Logical Property Group: <identifier>
```

where the identifier would typically be the most specific common shorthand of all the properties. (`margin`, `padding`, `border-style`, `border-color`, `inset`, etc.)

If we go with a statement, I would go with, e.g.:
> The longhands of 'border-style' form a logical property group.
so we don't have to list everything out. Easier to read, less likely to forget one.

**Comment by Loirooriol** [comment]
>     Logical Property Group: <identifier>

Works for me. But this should also appear in the associated physical properties, not just the logical ones. Because a logical property group has both.

> The longhands of 'border-style' form a logical property group.

According to https://drafts.csswg.org/css-backgrounds/#border-style, `border-style` is only a shorthand for the physical properties
CSS Logical seems to imply that it's a shorthand for both the physical and logical properties, but that part is considered unstable (#3030).
So I think this kind of statement that doesn't list everything is not completely clear.

**Comment by cdoublev** [comment]
Can you please tell me what would be the group identifier for `block-size`, `inline-size`, `width`, `height`? `size`?

**Comment by Loirooriol** [comment]
I don't think it matters. But [Blink](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/css/css_properties.json5;drc=7e07e7bfd0a21657dc9fd559f1e81b068aac2992;l=5774) and [WebKit](https://searchfox.org/wubkat/rev/355426c15db9b7f3fec2a3a35a743cdb5b1fc0d3/Source/WebCore/css/CSSProperties.json#1301) use `size` for `{block-size, inline-size, width, height}`. This is the list of their logical property groups names:
 - Blink: border, border-color, border-radius, border-style, border-width, contain-intrinsic-size, inset, margin, max-size, min-size, overflow, overscroll-behavior, padding, scroll-margin, scroll-padding, size, visited-border-color
 - WebKit: border-color, border-radius, border-style, border-width, contain-intrinsic-size, inset, margin, max-size, min-size, overscroll-behavior, padding, scroll-margin, scroll-padding, size

I don't think anybody is planning to do it soon. Better confirm the details with editors of the affected specs.

**Comment by cdoublev** [comment]
I was hoping to generate a representation of logical property groups/mappings with types similar to the following:

```
Group   => [Mapping]
Mapping => [PropertyName]
```

I do not need an explicit `Mapping` propdef but for example, [physical/logical `contain-intrinsic-*` longhands](https://drafts.csswg.org/css-sizing-4/#intrinsic-size-override) are defined in the same table, whereas [`scroll-padding-*` longhands](https://drafts.csswg.org/css-scroll-snap-1/#longhands) are separated in different definition tables for the different mappings.

I do not want to make a selfish request but please let me know if you think that some consistency should be applied. I would be happy to help.

**Comment by tabatkins** [comment]
I'm not sure I understand what you're asking about.

**Comment by cdoublev** [comment]
The procedures *set a CSS declaration* and *serialize a CSS declaration block* require to check whether a longhand 1. belongs to the same logical property group and 2. has a different mapping logic, than another longhand. A `Logical property group` propdef field allows to check 1 but to check 2, the logical property group, containing a set of longhand names, requires to be splitted in two subsets, corresponding to a mapping logic.

For example:

```js
const logicalPropertyGroups = [
  // ...
  'size': [
    ['block-size', 'inline-size'],
    ['height', 'width'],
  ],
  // ...
]
```

In specs, some longhands of the same logical property group and with the same mapping logic, are defined in the same definition table, which appears below a section title, eg. *Physical Longhands for `scroll-padding`*. 

`w3c/reffy` can take advantage of this grouping in the same table, and of this title, to define a `mappingLogic` field for the longhands defined in this table. 

But for some other longhands, like `contain-intrinsic-*`, both flow-relative and physical longhands are defined in the same table.

Solutions are:

  1. add a `Mapping logic` propdef field
  2. consistently define longhands with same/different mapping logic in same/different propdef table, below a title named after the corresponding mapping logic
  3. status quo: manually maintain the list/sublist of logical property groups and mapping logics

**Comment by tabatkins** [comment]
Ah, I see what you mean. Apologies, I wasn't fully up-to-date on the algorithms.

**Comment by cdoublev** [comment]
Now fixed in #7950. Now, at least, I can automatically watch for new logical properties and manually add them to the corresponding mapping logic subset.
